torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7581,6 +7581,32 @@ def sample_inputs_pdist(op_info, device, dtype, requires_grad, **kwargs):
         *[SampleInput(make_input((S, S)), kwargs=dict(p=p)) for p in (0.0, 1.0, 2.0, 10.0, float("inf"))],
     ]
 
+def sample_inputs_binary_cross_entropy(op_info, device, dtype, requires_grad, logits=False, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype)
+    make_prob = partial(make, low=0, high=1)
+
+    reductions = ("mean", "sum", "none")
+
+    shapes_and_kwargs = [
+        *[(shape, None) for shape in ((), (1,), (S,), (S, S), (S, S, S))],
+        *[((S, S), dict(reduction=reduction)) for reduction in reductions],
+        *[((S, S), dict(reduction=reduction, weight=make((S, S)))) for reduction in reductions],
```

Comment:
The pytorch documentation suggests that the weight must be 1D: https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html#torch.nn.BCELoss

Are the docs wrong?

Reply:
Looks like the docs may be wrong: #68780

Or we need to tighten up what is considered a valid shape for weight, but it seems like there are use cases outside of 1D weight of shape (N).